src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -140,6 +140,7 @@ private sealed class RunningInstance
 
             // Calling RunLoopStop multiple times SegFaults so protect the call to it
             private bool _stopping;
+            private object _stopLock = new object();
```

Comment:
> Would I still need the StopLock variable

No, I've just gotten into the habit of using a property so that I can name the object and make it clear that it's meant to be used for a lock, while also being able to change the actual object in the future, hence the:
```C#
private object StopLock => this;
```
(which is a property rather than a field)